frontend/public/co-fetch.js

```diff
@@ -164,7 +164,12 @@ export const coFetchJSON = (url, method = 'GET', options = {}, timeout) => {
 };
 
 export const coFetchText = (url, options = {}, timeout) => {
-  return coFetchCommon(url, 'GET', options, timeout);
+  return coFetchCommon(url, 'GET', options, timeout).then((response) => {
+    if (_.isObject(response)) {
+      return _.isEmpty(response) ? '' : JSON.stringify(response);
```

Comment:
This is odd because we're undoing what `coFetchCommon` has provided.
A better solution might be to split `coFetchCommon` into 2 functions such that one simply returns the response without calling `.text()` or `.json()` such that we use it here to return the text.

Reply:
The problem with coFetchCommon method is, when the `response.headers.get('Content-Length') === '0'` it always resolves an `{}` empty object, by conditionally resolving an  empty string or an empty object  based on `response.headers.get('Content-Type')` , we can avoid the undoing logic altogether. WDYT?